packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -296,6 +296,30 @@ enum SelectionEventType {
   ///
   /// Used by [SelectWordSelectionEvent].
   selectWord,
+
+  /// An event that extends the selection.
+  granularlyExtendSelection,
+
+  /// An event that extends the selection.
+  directionallyExtendSelection,
+}
+
+/// The unit of how selection handles move in text.
+///
+/// The [GranularlyExtendSelectionEvent] uses this enum to describe how
+/// [Selectable] should extend their selection.
+enum TextGranularity {
```

Comment:
CC @Renzo-Olivares since this reminds me of your work on longpress dragging by word and triple click dragging by line.